components/Tabs/src/TabBarView/MDCTabBarItem.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+#import "MDCTabBarItemCustomViewing.h"
+
+/**
+ A subclass of @c UITabBarItem that supports a custom view property.
+ */
+@interface MDCTabBarItem : UITabBarItem <MDCTabBarItemCustomViewing>
```

Comment:
This is intended to be public? Clients should use this? Should it have subclassing restricted?

Reply:
It is intended to be public, I can restrict subclassing on it.

Reply:
Do we expect other objects to conform to this protocol? If we just want to add a property to UITabBarItem do we really need this protocol to do that?

Reply:
I suspect most clients don't need more than this subclass. However, I didn't want to do `kindOfClass` checks, so the protocol makes the most sense for the implementation.  It also allows clients to use custom implementations without requiring subclassing of our class.

Reply:
Does that make sense? Do you think we shouldn't provide a custom subclass at all and just expect clients to write it (we'll still need subclasses for tests and example VCs).

Reply:
I don't quite follow yet. If clients don't use MDCTabBarItem will they use UITabBarItem? Or a different subclass if UITabBarItem? Or something that's not a UITabBarItem at all but conforms to MDCTabBarItemCustomViewing?

Reply:
They can use any instance or subclass of UITabBarItem for assigning items to the MDCTabBarView.  They can use ours if it's convenient or write their own subclass if that suits their needs better. Since one of our clients definitely needs a custom view for their own app, I think it made sense to provide a simple subclass out of the box, rather than have them write one.  I doubt many clients will conform to the protocol and not be able to use our subclass. However, since it's easier to work with protocols instead of expected subclasses, I went with the protocol route.

Reply:
Ok got it. Thanks for clarifying!